frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/EditFieldDrawer/edit-fieldtype/EditDate/EditDate.tsx

```diff
@@ -291,10 +291,20 @@ export const EditDate = ({ field }: EditDateProps): JSX.Element => {
           />
         </FormControl>
         {watchAddParticularDayRestriction ? (
-          <FormControl isRequired>
+          <FormControl isRequired isInvalid={!!errors.invalidDays}>
             <Controller
               control={control}
               name="invalidDays"
+              rules={{
+                validate: (val) => {
+                  const validDaysSet = new Set(val)
+                  return validDaysSet.has(
+                    InvalidDaysOptions.SingaporePublicHolidays,
+                  )
+                    ? val.length >= 2
+                    : val.length >= 1
+                },
+              }}
```

Comment:
what is this validation for? should we not need to just validate that the `val` array is non-empty?

Reply:
As discussed offline. I will make the necessary changes.